src/mcl.jl

```diff
@@ -0,0 +1,214 @@
+# MCL (Markov CLustering algorithm)
+
+#### Interface
+
+"""
+    `mcl()` result.
+"""
+immutable MCLResult <: ClusteringResult
+    mcl_adj::Matrix{Float64}    # final MCL adjacency matrix (equailibrium state matrix if converged)
+    assignments::Vector{Int}    # element-to-cluster assignments (n)
+    counts::Vector{Int}         # number of samples assigned to each cluster (k)
+    nunassigned::Int            # number of single elements not assigned to any cluster
+    iterations::Int             # number of elapsed iterations
+    rel_Δ::Float64              # final relative Δ
+    converged::Bool             # whether the procedure converged
+end
+
+""" _mcl_clusters(mcl_adj, allow_singles, zero_tol = 1E-20)
+
+    Extract clusters from the final MCL matrix `mcl_adj`.
+    Returns the vector of cluster indices for each element, the sizes of the
+    clusters and the number of the unassigned elements (`allow_singles` controls
+    whether 1-element clusters would be retained or collapsed into the set of
+    unassigned elements with the cluster index 0).
+
+    See also `mcl()`.
```

Comment:
Unindent all lines except for the function signature. But since this function isn't exported and isn't meant to be called directly, IMO it probably shouldn't have a docstring. You could convert the docstring to comments since the description is still useful to have.